lib/Reader/readerir.cpp

```diff
@@ -5381,11 +5519,15 @@ void GenIR::nop() {
   // Preserve Nops in debug builds since they may carry unique source positions.
   if ((JitContext->Flags & CORJIT_FLG_DEBUG_CODE) != 0) {
     // LLVM has no high-level NOP instruction. Put in a placeholder for now.
-    // We may need to pick something else that survives lowering.
-    Value *DoNothing = Intrinsic::getDeclaration(JitContext->CurrentModule,
-                                                 Intrinsic::donothing);
+    // This will survive lowering, but we may want to do something else that
+    // is cleaner
```

Comment:
I think we'll eventually want a high-level NOP that doesn't get removed and gets lowered to the right platform-specific encoding, so perhaps you can add a todo and an ISSUE to implement that in LLVM.